walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -33,6 +33,40 @@ private static boolean isAttended(Event event, MeetingRequest request) {
   }
 
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> freeTimes = new ArrayList<TimeRange>(); // Output list for possible meeting times.
+    Event[] eventsArray = events.toArray(new Event[events.size()]);
+    long duration = request.getDuration() / 30; // Meeting duration in number of half hour array slots.
+    
+    if (duration > 47) return freeTimes; // If the duration is out of bounds, we don't need to add any times.
+ 
+    boolean[] occupiedTimes = getOccupiedTimes(eventsArray, request);
+ 
+    int lowerBound = 0; // Start of Time Gap (first unoccupied slot past endPointer).
+    int upperBound = 0; // End of Time Gap (last unoccupied slot before first occupied slot after endpoint).
+ 
+    // Loop through array and find gaps between free slots.
```

Comment:
Instead of "array", it would be helpful to name it because you never know what might be added in the future. Also: did you mean to say "between occupied slots"?

Reply:
I've given the array a more applicable name and updated the comment here 96dd87c